keyboards/inett_studio/sqx/hotswap/hotswap.c

```diff
@@ -0,0 +1,178 @@
+    {1, C1_4,   C2_4,   C3_4},
+    {1, C1_5,   C2_5,   C3_5},
+    {1, C1_6,   C2_6,   C3_6},
+    {1, C1_7,   C2_7,   C3_7},
+    {1, C1_8,   C2_8,   C3_8},
+
+    //{1, C5_1,   C4_1,   C6_1},
+    //{1, C5_8,   C4_8,   C6_8},
+
+    {1, C9_1,   C8_1,   C7_1},
+    {1, C9_2,   C8_2,   C7_2},
+    {1, C9_3,   C8_3,   C7_3},
+    {1, C9_4,   C8_4,   C7_4},
+    {1, C9_5,   C8_5,   C7_5},
+    {1, C9_6,   C8_6,   C7_6},
+    {1, C9_7,   C8_7,   C6_6},
+    {1, C9_8,   C7_7,   C6_7},
+// right CB
+    {1, C1_9,   C3_10,  C4_10},
+    {1, C1_10,  C2_10,  C4_11},
+    {1, C1_11,  C2_11,  C3_11},
+    {1, C1_12,  C2_12,  C3_12},
+    {1, C1_13,  C2_13,  C3_13},
+    //{1, C1_14,  C2_14,  C3_14},
+    {1, C1_15,  C2_15,  C3_15},
+    {1, C1_16,  C2_16,  C3_16},
+
+    //{1, C5_9,   C4_9,   C6_9},
+    //{1, C5_16,  C4_16,  C6_16},
+
+    {1, C9_9,   C8_9,   C7_9},
+    {1, C9_10,  C8_10,  C7_10},
+    {1, C9_11,  C8_11,  C7_11},
+    {1, C9_12,  C8_12,  C7_12},
+    {1, C9_13,  C8_13,  C7_13},
+    {1, C9_14,  C8_14,  C7_14},
+    {1, C9_15,  C8_15,  C6_14},
+    {1, C9_16,  C7_15,  C6_15},
+};
+
+led_config_t g_led_config = {
+    {
+        {  0,       1,      2,      3,      4,      5,   6,     34,     35,     36, 37,   38,     39, 40},
+        {  8,       9,     10,     11,     12,     13,   7,     48,     41,     42, 43,   44,     45, 46},
+        {  26,     25,     18,     14,     15,     16,  17,     49,     50,     51, 52,   53, NO_LED, 47},
+        {  27, NO_LED,     28,     19,     20,     21,  22,     23,     56,     55, 57,   58,     59, 54},
+        {  32,     31,     30,     29, NO_LED, NO_LED,  24,     33, NO_LED, NO_LED, 60,   61,     62, 63},
+    },
+    {
+        {  0,  0},{ 16,  0},{ 32,  0},{ 48,  0},{ 64,  0},{ 80, 0},{ 96,  0},{ 96, 16},
+        {  0, 16},{ 16, 16},
+        { 32, 16},{ 48, 16},{ 64, 16},{ 80, 16},{ 64, 32},{ 80, 32},{ 96, 32},{112, 32},
+
+        { 48, 32},{ 48, 48},{ 64, 48},{ 80, 48},{ 96, 48},{112, 48},{112, 64},{ 32, 32},
+        { 16, 32},{ 16, 48},          { 32, 48},{ 64, 64},{ 40, 64},{ 20, 64},{  0, 64},
+
+        {112,  0},{128,  0},{144,  0},{160,  0},{176,  0},{192,  0},{208,  0},{224,  0},
+        {128, 16},{144, 16},{160, 16},{176, 16},{192, 16},{216, 16},{216, 32},{112, 16},
+
+        {128, 32},{144, 32},{160, 32},{176, 32},{192, 32},          {224, 48},{144, 48},
+        {128, 48},{160, 48},{176, 48},{192, 48},{160, 64},{180, 64},{200, 64},{224, 64}
+    },
+    {
+        1, 4, 4, 4, 4, 4, 4, 4,
+        1, 4,
+        4, 4, 4, 4, 4, 4, 4, 4,
+
+        4, 4, 4, 4, 4, 4, 4, 1,
+        1, 1,    4, 1, 1, 1, 1,
+
+        4, 4, 4, 4, 4, 4, 1, 1,
+        4, 4, 4, 4, 4, 1, 1, 4,
+
+        4, 4, 4, 4, 4,    1, 4,
+        4, 4, 4, 1, 1, 1, 1, 1,
+    }
+};
+
+#endif  //RGB_MATRIX_ENABLE
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed) {
+        switch(keycode) {
+        #ifdef RGBLIGHT_ENABLE
+            #ifdef RGB_MATRIX_ENABLE
```

Comment:
I'm not exactly happy with this implementation. 

I think that this may be better: 
```suggestion
#        ifdef RGB_MATRIX_DISABLE_KEYCODES
```

Or as separate keycodes.  Otherwise, you're preventing people from ever using these keys on this board.

Reply:
Yes, use custom keycodes was a better solution, but it's very hard to let the users to use the custom keycodes. Previously I was flooded by hundreds of questions for "what is custom keycodes ? " , thus why I choose the standard keycode that could be easier to be understood by the keyboard users.

Reply:
I have updated the keycodes definitions, please check.

Reply:
Just a heads up, the offical define name is `RGB_MATRIX_DISABLE_KEYCODES` and `RGBLIGHT_DISABLE_KEYCODES`.  

If you have both features enabled, the `RGB_*` keycodes will trigger both functions (eg, rgblight and rgb matrix). 

As for the questions about the keycodes ... it sounds like some documentation for the keyboard would go a long way.

Reply:
I have a question here, why not introduce some specific keycodes for the rgb matrix ? Are there any reasons to prevent the new keycodes ?

Reply:
@drashna The codes were updated, please check.

Reply:
Because, initially, you couldn't run rgb matrix and rgblight at the same time. This means that the keycodes could be reused in this way, without issues. 

However, between zvecr and myself, we've cleaned things up, so that you can run both features.  

However, this leaves the very tricky issue of how to handle the keycodes. And to do so that allows all 3 configurations.  (matrix, light, both).  Creating a new set of keycodes means that the existing keycodes don't work, and every keymap that uses them would have to be updated. Or aliases created for them.  Since it's kind of an edge case, especially as you can use rgb matrix to control ws2812 leds, and to have underglow LEDs, on the same string (like how the corne and massdrop boards do), it really hasn't been an issue.

In fact, until recently, I was the only one using both rgb matrix and rgblight at the same time, that I'm aware of.